modules/develop/pages/data-transforms.adoc

```diff
@@ -0,0 +1,251 @@
+:description: Learn how to run WebAssembly data transforms within Redpanda brokers.
+:page-aliases: reference:rpk/rpk-wasm/rpk-wasm.adoc, reference:rpk/rpk-wasm.adoc, reference:rpk/rpk-wasm/rpk-wasm-deploy.adoc, reference:rpk/rpk-wasm/rpk-wasm-generate.adoc, reference:rpk/rpk-wasm/rpk-wasm-remove.adoc, data-management:data-transform.adoc
+:latest-data-transforms-version: 0.0.0-20230830git604fcce
+
+Redpanda provides the framework to build and deploy inline transformations on data written to Redpanda topics, delivering processed and validated data to consumers in the format they expect. Data transforms let you run common data streaming tasks, like filtering, scrubbing, and transcoding, within the Redpanda broker. For example, you may have consumers that require you to remove credit card numbers or convert JSON to Avro. Data transforms can also interact with the Redpanda Schema Registry to work with encoded data types. Redpanda does this directly inside the broker, eliminating the need to manage a separate stream processing environment or use 3rd-party tools. 
+
+image::shared:wasm1.png[Data transforms in a broker] 
+
+Data transforms use a WebAssembly (Wasm) engine inside a Redpanda broker, allowing Redpanda to control the entire transform lifecycle. For example, Redpanda can stop and start transforms when partitions are moved or to free up system resources for other tasks. You can develop and manage data transforms with `rpk transform` commands.
+
+== Limitations
+
+- Transforms have no external access to disk or network resources. 
+- Only single record transforms are supported, not aggregations or joins. In other words, a Wasm function acts on a single record in an input topic.
+- Only a single output topic is supported.
+- Transforms have at-least-once delivery.
+- Because data transforms are powered by Wasm, transform functions can be authored in any language. However, currently libraries to author transforms are only available in xref:reference:data-transform-api.adoc[Golang^]. 
+
+== How data transforms work
+
+Within Repdanda, there's a single Raft controller that manages cluster information, including data transforms. On every shard (core), Redpanda knows what data transforms exist in the cluster, as well as metadata about the transform, such as input and output topics and environment variables. 
+
+Wasm is executed within a runtime in the broker. Data transforms take an input topic and map it to an output topic, running a virtual machine (VM) on the same shard as the input partition leader. Redpanda uses just-in-time (JIT) compilation, which compiles the function in memory, writes it to executable space, then runs the directly translated machine code. When it becomes leader of a given partition on the input topic, Redpanda spins up a Wasm VM using the JIT-compiled WebAssembly module. Redpanda writes the output, which may go to the same shard on the same machine, or to a different shard on the same machine, or to a different machine. 
```

Comment:
> Redpanda spins up a Wasm VM using the JIT-compiled WebAssembly module. Redpanda writes the output, which may go to the same shard on the same machine, or to a different shard on the same machine, or to a different machine. 

Somewhere inbetween here should probably say that we push records from the input partition into the Wasm VM

Reply:
please review the latest draft!